pydantic/deprecated/json.py

```diff
@@ -104,10 +113,16 @@ def custom_pydantic_encoder(type_encoders: Dict[Any, Callable[[Type[Any]], Any]]
         return pydantic_encoder(obj)
 
 
+@deprecated('timedelta_isoformat is deprecated.')
 def timedelta_isoformat(td: datetime.timedelta) -> str:
     """
     ISO 8601 encoding for Python timedelta object.
     """
+    warnings.warn(
+        'timedelta_isoformat is deprecated.',
+        DeprecationWarning,
+        stacklevel=2,
+    )
```

Comment:
Note: I'm guessing we can't create a decorator that applies both because static type checkers rely on the `@deprecated` decorator being used directly, right?

Reply:
right :cry: